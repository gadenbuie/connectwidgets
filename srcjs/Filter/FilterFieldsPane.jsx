import PropTypes from 'prop-types';
import FilterField from './FilterField';
import { FilterType } from './filterType';
import { FilterContext } from './stateUtils';
import './filterFieldsPane.scss';

function FilterFieldsPane({ isActive }) {
  const context = React.useContext(FilterContext);

  let paneClass = 'rscfilter-pane bg-white';

  if (isActive) {
    paneClass += ' rscfilter-pane--active';
  }

  return (
    <div className={paneClass}>
      <FilterField
        label="Owner"
        placeholder="Filter people"
        type={FilterType.owner}
      />
      <FilterField
        label="Type"
        placeholder="Select content types"
        type={FilterType.type}
      />
      <FilterField
        label="Tags"
        placeholder="Filter tags"
        type={FilterType.tag}
      />
      <div className="rscfilter-pane__actions border-top">
        <button
          className="rscfilter-pane__actions-reset btn btn-secondary"
          onClick={() => context.resetFilters()}
        >
          Reset
        </button>
      </div>
    </div>
  );
}

FilterFieldsPane.propTypes = {
  isActive: PropTypes.bool,
}

export default FilterFieldsPane;
