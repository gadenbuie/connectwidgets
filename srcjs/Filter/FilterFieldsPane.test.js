import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FilterFieldsPane from './FilterFieldsPane';

const fakeContext = { resetFilters: jest.fn() };
jest.spyOn(React, 'useContext').mockReturnValue(fakeContext);

const mkWrapper = () => shallow(<FilterFieldsPane isActive={false} />);

describe('<FilterFieldsPane />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('renders properly', () => {
    const wrapper = mkWrapper();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('can call to reset filters state', () => {
    const wrapper = mkWrapper();
    expect(fakeContext.resetFilters).not.toHaveBeenCalled();
    wrapper.find('button').simulate('click');
    expect(fakeContext.resetFilters).toHaveBeenCalled();
  });
});
