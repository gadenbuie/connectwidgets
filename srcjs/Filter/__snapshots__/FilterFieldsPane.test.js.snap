// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterFieldsPane /> renders properly 1`] = `
<div
  className="rscfilter-pane bg-white"
>
  <FilterField
    label="Owner"
    placeholder="Filter people"
    type="owner"
  />
  <FilterField
    label="Type"
    placeholder="Select content types"
    type="type"
  />
  <FilterField
    label="Tags"
    placeholder="Filter tags"
    type="tag"
  />
  <div
    className="rscfilter-pane__actions border-top"
  >
    <button
      className="rscfilter-pane__actions-reset btn btn-secondary"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</div>
`;
